src/coreclr/jit/gentree.h

```diff
@@ -8039,12 +8039,12 @@ inline bool GenTree::IsMultiRegNode() const
         return true;
     }
 #endif // FEATURE_MULTIREG_RET
-#if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
-    if (OperIs(GT_HWINTRINSIC))
+
+    if (OperIsHWIntrinsic())
     {
-        return (TypeGet() == TYP_STRUCT);
+        return TypeIs(TYP_STRUCT);
```

Comment:
I wonder how well this will hold in the future? That is, are we expecting things that return `TYP_STRUCT `to always be multi-reg?

There are cases like say `System.Half` where we'll eventually need to add support and we'll need to treat it as `TYP_HALF` or recognize it some other way if we don't want it to be an issue.

I wonder if we should track this as a `flag` rather than by type just to be safe?

Reply:
Ah, looks like we have a flag. Maybe we should assert it or use it here instead?

Reply:
Sure, I can use a flag here.